src/lib.rs

```diff
@@ -1390,6 +1399,21 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 				continue;
 			}
 
+			if let Some(first_route_hop) = path.hops.first() {
+				if let Some(first_hop) = first_hops
+					.iter()
+					.find(|h| h.short_channel_id == Some(first_route_hop.short_channel_id))
+				{
+					let path_value = path.final_value_msat() + path.fee_msat();
+					if first_hop.next_outbound_htlc_limit_msat
+						< path_value * self.config.probing_liquidity_limit_multiplier
```

Comment:
If a user has one channel and is trying to make a payment that spends greater than half of its value, won't this still be a problem if the payment is split equally along three paths?

Say that `next_outbound_htlc_limit_msat` is initially `100` and the payment is split equally along paths each with a `path_value` of `19` for a total payment value of `57`. All three paths would be probed as their value is under 1/3 `next_outbound_htlc_limit_msat` even when adjusting for the previously probed path, but we are taking over half the channel liquidity. Worse case scenario is all three probes don't come back and we can't make the payment because our only channel's liquidity isn't enough for the full payment.

Should we instead keep a running sum of `path_value` for paths using the same first hop, and probe until we exceeded the limit with a fixed `next_outbound_htlc_limit_msat`?

Reply:
It would depend on the scorer used, it's state, which channels are probed, and the values for each path. Yes, for `ProbabilisticScorer`, as it's trying to learn channel liquidity balances and how often they fall within a given bucket.

As for the follow-up payment, if a probe fails for one of the paths, then we'll know to avoid a channel or trying splitting differently.